unified-scheduler-pool/src/lib.rs

```diff
@@ -117,7 +141,46 @@ where
             weak_self: weak_self.clone(),
             next_scheduler_id: AtomicSchedulerId::default(),
             _phantom: PhantomData,
-        })
+        });
+
+        let cleaner_main_loop = || {
+            let weak_scheduler_pool = Arc::downgrade(&scheduler_pool);
+
+            move || loop {
```

Comment:
See comment below, just create the looping closure directly instead of a closure creating closure:
```suggestion
        let cleaner_main_loop = {
            let weak_scheduler_pool = Arc::downgrade(&scheduler_pool);
            move || loop {
```

Reply:
done: 1db0675acbb21c309dee72d9ce8aea1c23b5faaa